scopes/ui-foundation/use-box/menu/menu.tsx

```diff
@@ -1,55 +1,42 @@
-import React, { useState } from 'react';
+import React, { useState, useMemo, ReactNode } from 'react';
 import { Icon } from '@teambit/evangelist.elements.icon';
-import { BitInfo } from '@teambit/ui-foundation.ui.use-box.bit-info';
-import { Registry } from './registry';
-import { Install } from './install';
-import { Import } from './import';
+import { Tab } from '@teambit/ui-foundation.ui.use-box.tab';
+
 import styles from './menu.module.scss';
-import { Tabs } from './tabs';
-import { Elements } from './elements';
+
+// TODO where should we place this type? its being created in component ui runtime as well
+export type ConsumeMethod = {
+  Title?: ReactNode;
+  Component?: ReactNode;
+  order?: number;
+};
 
 export type MenuProps = {
   /**
-   * package link to be copied
+   * consume methods to be displayed in the menu.
    */
-  packageName: string;
+  methods: ConsumeMethod[];
 
-  /**
-   * Elements url
-   */
-  elementsUrl?: string;
-  /**
-   * import link to be copied
-   */
-  componentId: string;
-  /**
-   * registry link to be copied
-   */
-  registryName: string;
   /**
    * component name to be presented
    */
   componentName: string;
 } & React.HTMLAttributes<HTMLDivElement>;
 
-export function Menu({ packageName, componentId, registryName, componentName, elementsUrl, ...rest }: MenuProps) {
-  const [activeTab, setActiveTab] = useState('bit');
-  const [activeRegistry, setActiveRegistry] = useState<string | undefined>(undefined);
-
-  if (activeRegistry === 'import') {
-    return <BitInfo prevTab={activeTab} setActive={() => setActiveRegistry(undefined)} />;
-  }
+export function Menu({ methods, componentName, ...rest }: MenuProps) {
+  const [activeTab, setActiveTab] = useState(0);
+  const content = useMemo(
+    () =>
+      methods.flat().sort((a, b) => {
```

Comment:
we dont have polyfills for flat. use lodash.flatten, or maybe even lodash.orderBy